### ISSUE TITLE
can not click on namespace std and it's classes

### ISSUE THREAD

**Issue by paule32** [doxygen/doxygen#10525]
I would like to document the STL.
So, I include ./doc/std/string.
The namespace will be included in the CHM Project.
But I can not click on the several "std::string foo(void) ..." functions. (png 2)
Nor, I can not click on the namespaces (png 3)

I expect, that I can click on "std" to come to namespace list
I expect, that I can click on "string" to come to this class/template
I expect, that I can click on namespace (png 3) to come to the namespace.

I add a .cc file with input text, and 3 png picture, which I have marked with small comment
in the attached file below:

[packed.zip](https://github.com/doxygen/doxygen/files/13800099/packed.zip)

**Comment by albert-github** [comment]
In your used Doxyfile you set `BUILTIN_STL_SUPPORT    = YES` which means use the internal list and as the internal list doesn't have documentation this is explicitly disabled for `std`.

You want to explicitly document `std::string` so you will have set set `BUILTIN_STL_SUPPORT    = NO`.

I think this problem is usage.

**Comment by paule32** [comment]
and how can I set the color of the return types:  std::string foo(void) ?

**Comment by albert-github** [comment]
Where do you want to set the color exactly, as I see `std::string` on multiple places?

**Comment by paule32** [comment]
yes, correct. on multiple places:
- where the definition for a function is inner/outer space of a class/namespace.
- and/or in the description where the function will be describe.

**Comment by albert-github** [comment]
Please illustrate with images so it is clear to all (it is not clear to me).

**Comment by paule32** [comment]
The Arrow's marks the (return) Type, which I would like customize with other color.

![grafik](https://github.com/doxygen/doxygen/assets/3682603/7ae46dee-100d-46cb-a90b-1bcb5eba9e1b)

**Comment by albert-github** [comment]
This has to be done with with the css, like:
```
.memItemLeft a {color:red }
```
this will set the color to red, but as soon as the link is finished it will have a grayish color, so also the part `a:visited` has to be set like:
```
.memItemLeft a:visited {color:yellow }
```
there are other settings possible as well like `hover`, see the css documentation.
Note this will be done for all return types on this place, when you only want `std::string` maybe something is possible, check the css documentation or the Internet.

**Comment by paule32** [comment]
Thank you very much !!!
It could be a good idea, to list the CSS Styles for what Style stands.
Not that I don't know CSS, and don't have skill's to change stylesheet's - but when nobody knows which style ID (#) or CLASS (.) stands for - then it is not easy to found out.
If there a such list, It would be nice to see, that you refer me to the right direction.
But, indeed, you/me can ask Question's there in this Channel/Issue Forum, where the people stuckle on ...

So again, many thanks, and btw.: A happy new Year !!!

**Comment by albert-github** [comment]
There is no such list and there is no intention to create one (all items should be available in the standard `doxygen.css`).

It is rather "easy" to figure this out either:
- read the resulting html file

or
- use the "inspect(Q)" that you get on Firefox when selecting a word and right click on it (In Edge probably same procedure).

For both of course some knowledge of CSS is necessary as well as some searching.

**Comment by paule32** [comment]
okay. I get it done.
But, is it possible by doxygen to parse the CSS files ?
So my idea was, when doxygen have parsed a CSS Style "display: none;" (and no one styles are present) the HTML stuff will be remove in HTM/CHM Project.
This will then much smaller CHM Files, because the value of HTML (maybe the header navigation is not see (display:none;) consume zero space instead much HTML on each Page.

What did you think about, Albert ?

**Comment by albert-github** [comment]
The HTML files as used by the CHM are the same as the normal HTML files to be viewed with a browser.
It might be that when having some extra intelligence in doxygen some parts could be skipped in the HTML output, but what would happen with the LaTeX / RTF  / ... output? 
Furthermore when the user, of the HTML files, changes settings later on (e.g. before sending the files to a HTML server) this, at that moment maybe relevan code might not be present anymore.
Doxygen does not interpret the css files they are just present and considered templates for the output.

**Comment by paule32** [comment]
But I could a Filter for SED ?

**Comment by albert-github** [comment]
When I understand it you would like to apply `sed` on the output? This would be a postprocessing step for the user and could be done, afterwards the `chm` file could be recreated "by hand".

**Comment by albert-github** [comment]
Not completely.
- doxygen creates the HTML file and also the `hhc`, `hhp` and `hhk` files
- when finished with creating all these file the chm file is created by means of calling the HTMLHelp compiler

So when there would be a postprocessing step, but the question is always where would this be put exactly, the syeps could be
- doxygen creates the HTML file and also the `hhc`, `hhp` and `hhk` files
- when finished with creating all these file the postprocessing could start
- the chm file is created by calling the HTMLHelp compiler

By not setting `HHC_LOCATION` the `hhc`, `hhp` and `hhk` filesare stil created but the HTMLHelp compiler is not started.
When using e.g. CMake one could use the workflow
- run dooxygen (without setting `HHC_LOCATION`)
- run the postprocessing
- run the HTMLHelp compiler

> Would be nice, that give a TAG, that can be used, to "only" create the HTML + CHM Project files.

So in other words this "tag" is available y not setting `HHC_LOCATION`.

**Comment by paule32** [comment]
Thanks, you made my day :-)
